### DISCUSSION TITLE
Compilation instructions for Ubuntu - missing dependencies

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1143] [Team Posts]
Someone I recommended MusE to complained that the compilation instructions for Ubuntu lead to error. We got out that there are missing dependencies in the installation list for apt-get in the README.
One of them was "liblilv-dev", the other one I forgot but will try to get out (putting it down so I don't forget).

More feedback I got:
- broken web
- .deb package or AppImage would be good, many people don't like to compile
- setting left marker with middle mouse button clumsy (you can map it to Ctrl+right click, but a symmetrical function would be better, like setting left marker with Ctrl+left mouse and the right one with Ctrl+right mouse).
- using sf3 sound fonts with MESS loader crashes (I addressed that already in Issues).

**Comment by spamatica** [comment]
Do you know which release of ubuntu?

As for the rest of the problems I pretty much agree.
sf3 I don't know anything about.. I'll have to try if it works for me

**Comment by kybos** [comment]
It was 19.10 (Ubuntu studio), but I remember having the same problem on my Mint (based on Ubuntu 18.04).

**Comment by kybos** [comment]
The other missing dep was liblrdf0-dev.
Could we also delete README.de? It's hopelessly outdated so it could lead to confusion as well.

**Comment by spamatica** [comment]
Deleted it, we have enough trouble keeping one README file up to date :P

**Comment by kybos** [comment]
Should we add liblilv-dev and liblrdf0-dev to the apt-get install list in README or is there a reason they are not in?

**Comment by spamatica** [comment]
Go ahead and update it if that is all that is needed.
These dependencies were quite recently added, we had our own versions of
these libs before but I think Tim changed it since it turned out hard to
maintain.

Den mån 16 dec. 2019 kl 19:54 skrev kybos <notifications@github.com>:

**Comment by terminator356** [comment]
Deb: Isn't this what we're looking for?
[https://packages.debian.org/buster/muse](url)

liblilv-dev yes. Required by our LV2 module.
Also sord and lv2, as mentioned in the README LV2 section.
Until recently we had all those libraries internally in case users didn't have them or whatever.
But I removed all that support. So users must install them separately.

librdf: No, not required at the moment. You can leave it out. It basically does nothing for now.
Plans were to use it when scanning plugins. But it's basically unused ATM.

**Comment by terminator356** [comment]
Let's get rid of:
NEWS, qt4migration.log
And muse.pro and .prj look severely out of date. What's that prj file for? What's PRCS?
I'm not sure anyone would need them since QtCreator now supports cmake files.

**Comment by kybos** [comment]
> Deb: Isn't this what we're looking for?
> https://packages.debian.org/buster/muse

It's the 3.0.2 release which is now available in most Debian-based distros.
The .deb or appimage idea went more in direction of more up-to-date (pre-)releases. Typically it takes very long to get current app release into Debian/Ubuntu., especially the Ubuntu LTS releases...

**Comment by spamatica** [comment]
Den mån 16 dec. 2019 kl 20:43 skrev kybos <notifications@github.com>:

> Deb: Isn't this what we're looking for?
> https://packages.debian.org/buster/muse
>
> It's the 3.0.2 release which is now available in most Debian-based distros.
> The .deb or appimage idea went more in direction of more up-to-date
> (pre-)releases. Typically it takes very long to get current app release
> into Debian/Ubuntu., especially the Ubuntu LTS releases...
>
The idea with appimage is pretty good, providing a distribution independent
version of applications.
There is also flatpak and there is actually a muse build for flatpak:
https://flathub.org/apps/details/org.musesequencer.Muse3

But on my system it does not seem to work, I haven't researched why but I'm
thinking it might be too much to ask of an application that needs rather
low level access to other parts of your operating system.. jack, alsa etc.
Linux is dependency sensitive as it is...

**Comment by terminator356** [comment]
Tried flatpak. The site warned that I can only install via the GUI that comes with my distro.
So I set it up, clicked install, and it got about quarter through and it said:
"Aborted due to error".

**Comment by kybos** [comment]
I am a little sceptical about hybrid solutions like flatpak and snap, never liked them (they install kind of subsytems on which the apps depend). AppImage is different in that it is truly self-contained. QTractor uses it as well for the distribution of new releases, so that should be proof enough that it also works for resource-critical applications.

**Comment by kybos** [comment]
> And muse.pro and .prj look severely out of date. What's that prj file for? What's PRCS?
> I'm not sure anyone would need them since QtCreator now supports cmake files.

PRCS is some obscure versioning system (and prj its file extension). It's outdated (same as the .pro file for QtCreator), so hardly of any use.

**Comment by spamatica** [comment]
> I am a little sceptical about hybrid solutions like flatpak and snap, never liked them (they install kind of subsytems on which the apps depend). AppImage is different in that it is truly self-contained. QTractor uses it as well for the distribution of new releases, so that should be proof enough that it also works for resource-critical applications.

I'm not convinced there is a difference in the approaches, but I know too little and if it works for QTractor that is good news!

**Comment by spamatica** [comment]
> PRCS is some obscure versioning system (and prj its file extension). It's outdated (same as the .pro file for QtCreator), so hardly of any use.

I'm all for cleaning these old files out.
